### PR TITLE
CoreDNS - Connect to external kubernetes using TLS, enable TCP and UDP for dns://

### PR DESCRIPTION
**Pull request by oliverisaac**
**What this PR does / why we need it**: 

- In order to get CoreDNS to connect to an external GKE instance you need to use the tls key/cert/cacert configuration. This requires setting up a secret with the appropriate files then pointing CoreDNS at that secret.
- Also, and related to this entire adventure I've been on, CoreDNS supports both UDP and TCP out of the box if the protocol is dns://. The current implementation doesn't set this up and this causes programs which try to use TCP to fail. This PR enables TCP and UDP when using either dns:// as the protocol or when no protocol is set (which defaults to dns://)

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/coredns/values.yaml [hunk 1]
```diff
@@ -64,6 +64,24 @@ servers:
...
+#
+#  Example kube config to use endpoint:
+#  servers:
+#    plugins:
+#    - name: kubernetes
+#      parameters: primary.local
+#      configBlock: |-
+#        endpoint https://1.2.3.4
+
+kubernetesTls:
```
Comment (paulczar) [inline stable/coredns/values.yaml]: for readability this should be `kubernetesTLS:`

File: stable/coredns/templates/secret.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{{- if .Values.kubernetesTls.enabled }}
```
Comment (paulczar) [inline stable/coredns/templates/secret.yaml]: I'd like to see an option for the user to specify a secret they've already created, that way they can elect to not store the secrets in helm values.

**Review by paulczar — CHANGES_REQUESTED** [review]

File: stable/coredns/templates/service.yaml [hunk 3]
```diff
@@ -26,4 +26,4 @@ spec:
   {{- end }}
   ports:
 {{ include "coredns.servicePorts" . | indent 2 -}}
-  type: {{ default "ClusterIP" .Values.serviceType }}
+  type: {{ default "ClusterIP" .Values.service.type }}
```
Comment (paulczar) [inline stable/coredns/templates/service.yaml:32]: This is a breaking change, so needs to be bumped to v 2.     also I don't see a change to values.yaml to represent the change.

**Review by paulczar — CHANGES_REQUESTED** [review]

File: stable/coredns/templates/service.yaml (hunk 3 above)
Comment (oliverisaac) [reply stable/coredns/templates/service.yaml:32]: I've updated values.yaml to reflect the change as well as bumping version number to 2.0.0

**Review by paulczar — APPROVED** [review]

**Comment by paulczar** [comment]
So this looks good to me, but I don't really understand the cause of the helm lint error

**Pull request closed without merging**